File: CPSC_231_F2018/assignment_2/java_bonus/HaohuShen_p5.java
/**
 * This program simulates a computer's turn of pig.
 * It will not stop rolling until accumulating 20 or more points in the turn.
 *
 * @author Haohu Shen
 * @version 1.8.0_161, 09/24/2018
 */

import java.util.concurrent.ThreadLocalRandom;

public class HaohuShen_p5 {

    // constants
    private static final int MIN_VAL_OF_A_DIE = 1;
    private static final int MAX_VAL_OF_A_DIE = 6;
    private static final int PIG_OUT = 1;
    private static final int MAX_SCORE = 20;

    public static void main(String[] args) {

        //simulation
        int turn_score = 0;
        while (true) {
            int outcome = ThreadLocalRandom.current().nextInt(
                    MIN_VAL_OF_A_DIE, MAX_VAL_OF_A_DIE + 1);
            System.out.println(" - rolled a " + outcome);
            if (outcome == PIG_OUT) {
                turn_score = 0;
                System.out.println("Pigged out!");
                break;
            }
            turn_score += outcome;
            if (turn_score >= MAX_SCORE) {
                break;
            }
        }

        // output
        System.out.println("Turn score = " + turn_score);
    }
}

File: CPSC_231_F2018/assignment_2/HaohuShen_p5.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-

"""
This program simulates a computer's turn of pig. It will not stop rolling
until accumulating 20 or more points in the turn.

    Author: Haohu Shen
    Date:   September 2018
"""

import random

# constants
MIN_VAL_OF_A_DIE = 1
MAX_VAL_OF_A_DIE = 6

PIG_OUT = 1
MAX_SCORE = 20

if __name__ == "__main__":

    # initialization
    random.seed()

    # simulation
    turn_score = 0
    while True:
        outcome = random.randint(MIN_VAL_OF_A_DIE, MAX_VAL_OF_A_DIE)
        print(" - rolled a " + str(outcome))
        if outcome == PIG_OUT:
            turn_score = 0
            print("Pigged out!")
            break
        turn_score += outcome
        if turn_score >= MAX_SCORE:
            break

    # output
    print("Turn score = " + str(turn_score))
